compiler/x/runtime/X86Runtime.hpp

```diff
@@ -71,4 +74,30 @@ inline bool jitGetCPUID(TR_X86CPUIDBuffer* pBuffer)
       }
    }
 
+inline bool InterlockedCompareAndSwap(volatile uint32_t* ptr, uint32_t old_val, uint32_t new_val)
```

Comment:
I'm not overly keen on the `Interlocked` prefix to this function despite the fact that the Microsoft compiler uses it.  My main objection is that it's a term that is foreign to the Testarossa compiler technology and hasn't been used before in this context.  `Atomic` is probably more appropriate (e.g., `AtomicCompareAndSwap`).

Reply:
Good suggestion, I've updated the name.